### ISSUE TITLE
Image display issue

### ISSUE THREAD

**Issue by lionel008** [astarasikov/osx-opengl-ffmpeg-test#1]
hello image output is upside down , any idea?

**Comment by astarasikov** [comment]
You are right, the image is upside down. You will need to swap the texture coordinates or rectangle coordinates. For example, swap 0 for 1 and 1 for 0 in the second column in lines 24-27 (where there is the comment "texture coordinates").

I have recently made changes to build with the latest ffmpeg (which renamed the frame alloc/free functions) on osx 10.12, but I have not pushed them yet. I'll see if I can push them.

If you seriously consider using this code, I would recommend also adjusting the coefficients in the GLSL shader for the correct YUV2RGB conversion, but it should provide a good starting point.

**Comment by lionel008** [comment]
Hi thanks for replay , already tried that but still something wrong , please have a look 


Have you got direct mail?

**Comment by astarasikov** [comment]
It looks like your video is also flipped horizontally. Can you check that in a video player because as far as I remember, that was not happening for me?

My email is my alexander.tarasikov@gmail.com

**Comment by lionel008** [comment]
Hi i already check in vlc and other players and it is fine , also , if i convert AVFRAME to NSIMAGE , it is fine as well , think something to do with opengl?
